proposals/NNNN-codable-error-printing.md

```diff
@@ -0,0 +1,160 @@
+
+**Note 1:** this proposal is _not_ intended to specify an exact output format. The above is provided as an example, and is not a guarantee of current or future behavior. You are still free to inspect the contents of thrown errors directly if you need to detect specific problems.
+
+**Note 2:** the output could be further improved by modifying `JSONDecoder` to write a better debug description. See [Future Directions](#future-directions) for more.
+
+## Detailed design
+
+```swift
+@available(SwiftStdlib 6.2, *)
+extension EncodingError: CustomDebugStringConvertible {
+  public var debugDescription: String {...}
+}
+
+@available(SwiftStdlib 6.2, *)
+extension DecodingError: CustomDebugStringConvertible {
+  public var debugDescription: String {...}
+}
+```
+
+## Source compatibility
+
+The new conformance changes the result of converting an `EncodingError` or `DecodingError` value to a string. This changes observable behavior: code that attempts to parse the result of `String(describing:)` or `String(reflecting:)` can be misled by the change of format.
+
+However, the documentation of these interfaces explicitly state that when the input type conforms to none of the standard string conversion protocols, then the result of these operations is unspecified.
+
+Changing the value of an unspecified result is not considered to be a source incompatible change.
+
+## ABI compatibility
+
+The proposal conforms two previously existing stdlib types to a previously existing stdlib protocol. This is technically an ABI breaking change: on ABI-stable platforms, we may have preexisting Swift binaries that implement a retroactive `CustomDebugStringConvertible` conformance, or binaries that assume that the existing error types do _not_ conform to the protocol.
+
+We do not expect this to be an issue in practice. The above checks represent edge cases, since checking an arbitrary error for conformance to `CustomDebugStringConvertible` at run-time seems unlikely. And if someone has added a conformance, when they re-build with a stdlib that includes this conformance, they will get a build-time warning.
+
+## Implications on adoption
+
+### Conformance to `CustomDebugStringConvertible`
+
+The conformance to `CustomDebugStringConvertible` is not backdeployable. As a result, code that runs on ABI-stable platforms with earlier versions of the standard library won't output the new debug descriptions.
+
+### `debugDescription` Property
+
+The change to the debug output is purely cosmetic: it improves readability, but does not change the information contained in the message. For this reason, we think it is not worthwhile to backport the `debugDescription` property itself. Since the conformance to `CustomDebugStringConvertible` is not back-deployable, a backport of property itself would be of limited utility anyway.
```

Comment:
Whether something _can_ be backdeployed is an implication on adoption that is appropriately part of the proposal, but it's outside of the scope of Swift Evolution whether something _actually will_ be backdeployed on any particular platform since it's ultimately up to the platform owner. So you could make this even shorter and say something like, the implementation of `debugDescription` itself is possible to backdeploy but would be on its own be of limited utility.